app/src/common/shared/org/mozilla/vrbrowser/browser/Media.java

```diff
@@ -132,27 +139,21 @@ public void onPlaybackStateChange(MediaElement mediaElement, int playbackState)
         } else if (playbackState == MediaElement.MEDIA_STATE_ENDED) {
             mEnded = true;
         }
-        if (mDelegate != null) {
-            mDelegate.onPlaybackStateChange(mediaElement, playbackState);
-        }
+        mMediaListeners.forEach(listener -> listener.onPlaybackStateChange(mediaElement, playbackState));
     }
 
     @Override
     public void onReadyStateChange(MediaElement mediaElement, int readyState) {
         mReadyState = readyState;
-        if (mDelegate != null) {
-            mDelegate.onReadyStateChange(mediaElement, readyState);
-        }
+        mMediaListeners.forEach(listener -> listener.onReadyStateChange(mediaElement, readyState));
```

Comment:
Iterating the listeners array can be risky if a listener unregister in on of the calls. Consider creating a copy before notification or using `CopyOnWriteArrayList` or similar ArrayList

Reply:
Right. Didn't know about `CopyOnWriteArrayList` though. Thanks for that!